marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java

```diff
@@ -76,11 +84,85 @@
     @RequestMapping(value = "/{modelName}/entityTypes", method = RequestMethod.PUT)
     @ApiImplicitParam(required = true, paramType = "body", dataType = "ModelDefinitions")
     public ResponseEntity<Void> updateModelEntityTypes(@ApiParam(hidden = true) @RequestBody JsonNode entityTypes, @PathVariable String modelName) {
+        // update the model
         newService().updateModelEntityTypes(modelName, entityTypes);
+
+        //deploy updated configs
+        deployModelConfigs();
+
         return new ResponseEntity<>(HttpStatus.OK);
     }
 
 
+    void deployModelConfigs() {
+        ManageClient manageClient = hubClientProvider.getHubClient().getManageClient();
+        DatabaseClient finalDatabaseClient = hubClientProvider.getHubClient().getFinalClient();
+        DatabaseClient stagingDatabaseClient = hubClientProvider.getHubClient().getStagingClient();
+
+        JsonNode modelConfigNode = newService().generateModelConfig();
+
+        // Deploy search options
+        String explorerOptions = modelConfigNode.get("search-options").get("explorer").asText();
+        String defaultOptions = modelConfigNode.get("search-options").get("default").asText();
+        Map<String, DatabaseClient> clientMap = new HashMap<>();
+        clientMap.put("staging", stagingDatabaseClient);
+        clientMap.put("final", finalDatabaseClient);
+        clientMap.forEach((databaseKind, databaseClient) -> {
+            QueryOptionsManager queryOptionsManager = databaseClient.newServerConfigManager().newQueryOptionsManager();
+            queryOptionsManager.writeOptionsAs("exp-" + databaseKind + "-entity-options", Format.XML, explorerOptions);
+            queryOptionsManager.writeOptionsAs(databaseKind + "-entity-options", Format.XML, defaultOptions);
+        });
+
+        // Deploy protected-paths
+        modelConfigNode.get("protected-paths").forEach(jsonNode -> manageClient.postJson("/manage/v2/protected-paths", jsonNode.toString()));
+
+        // Deploy query-rolesets
+        try {
+            modelConfigNode.get("query-rolesets").forEach(jsonNode -> manageClient.postJson("/manage/v2/query-rolesets", jsonNode.toString()));
```

Comment:
You need to do a try/catch for each individual postJson call. While we only expect one query roleset for now (pii-reader), there could be more in the future. And if the first one fails because it already exists, we still want to try to deploy the others.

Reply:
Good catch. Updated in here b6777af